### PR TITLE
[FIXED JENKINS-51336] Hide parser details by default

### PR DESCRIPTION
**Pull request by pjdarton**
If we have more than one or two parsers defined in the plugin then the main Jenkins configuration UI page gets very crowded and difficult to navigate.
The plugin should hide the details unless the user actually needs to see the details.

This change hides all parsers until the "Parsers..." button is clicked on.
Individual parsers show only their name until the "Details..." button is clicked on.

i.e. Much like [JENKINS-20909](https://issues.jenkins-ci.org/browse/JENKINS-51336), but for the warnings-plugin.

### CODE REVIEW

File: src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly [hunk 1]
```diff
@@ -1,27 +1,27 @@
 <?jelly escape-by-default='true'?>
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly]: This line had a unix EOL code (\n) at its end.
All the other lines in this file had MSDOS EOL codes (\r\n) at their ends.
...except the last line, which had no EOL code.

File: src/main/resources/hudson/plugins/warnings/WarningsPublisher/global.jelly [hunk 2]
```diff
@@ -1,18 +1,18 @@
-<?jelly escape-by-default='true'?>
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/WarningsPublisher/global.jelly]: All the lines in this file had MSDOS EOL codes (\r\n) at their ends.
...except the last line, which had no EOL code.

File: src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly [hunk 3]
```diff
@@ -4,21 +4,23 @@
...
-  <f:entry field="regexp" title="${%Regular Expression}" description="${%regexp.description}">
-    <f:textbox />
-  </f:entry>
-  <f:entry field="script" title="${%Mapping Script}" description="${%script.description}">
-    <f:textarea />
-  </f:entry>
-  <f:entry field="example" title="${%Example Log Message}" description="${%example.description}">
-    <f:textarea />
-  </f:entry>
+  <f:advanced title="${%Details}" align="left">
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly]: Added this line.

File: src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly [hunk 4]
```diff
@@ -4,21 +4,23 @@
...
+    </f:entry>
+    <f:entry field="regexp" title="${%Regular Expression}" description="${%regexp.description}">
+      <f:textbox />
+    </f:entry>
+    <f:entry field="script" title="${%Mapping Script}" description="${%script.description}">
+      <f:textarea />
+    </f:entry>
+    <f:entry field="example" title="${%Example Log Message}" description="${%example.description}">
+      <f:textarea />
+    </f:entry>
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly]: Indented these by two spaces.

File: src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly [hunk 5]
```diff
@@ -4,21 +4,23 @@
...
+  </f:advanced>
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly]: ...and added this line.

File: src/main/resources/hudson/plugins/warnings/GroovyParser/config_de.properties [hunk 6]
```diff
@@ -13,6 +13,7 @@ example.description=Beispielmeldung, die automatisch mit den oben angegebenen We
 Link\ name=Name des Ergebnislinks
 Trend\ report\ name=Name des Trends
 Name=Name
+Details=Einzelheiten
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/GroovyParser/config_de.properties]: Translation courtesy of Google Translate.

File: src/main/resources/hudson/plugins/warnings/GroovyParser/config_ja.properties [hunk 7]
```diff
@@ -1,4 +1,5 @@
 Name=\u540d\u524d
+Details=\u8a73\u7d30
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/GroovyParser/config_ja.properties]: Translation courtesy of Google Translate, verified by a colleague of mine.

File: src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly [hunk 8]
```diff
@@ -1,27 +1,29 @@
...
-    </div>
-  </f:entry>
-</j:jelly>
\ No newline at end of file
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define"
+  xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+  <f:entry field="name" title="${%Name}" description="${%name.description}">
+    <f:textbox />
+  </f:entry>
+  <f:advanced title="${%Details}" align="left">
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/GroovyParser/config.jelly]: Added this line.

File: src/main/resources/hudson/plugins/warnings/WarningsPublisher/global.jelly [hunk 9]
```diff
@@ -1,18 +1,20 @@
...
- </f:section>
-</j:jelly>
\ No newline at end of file
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:u="/util"
+         xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+  <f:section title="${%Compiler Warnings}">
+      <j:choose>
+        <j:when test="${descriptor.canEditParsers()}">
+          <f:advanced title="${%Parsers}" align="left">
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/WarningsPublisher/global.jelly]: Added this line.

File: src/main/resources/hudson/plugins/warnings/WarningsPublisher/global.jelly [hunk 10]
```diff
@@ -1,18 +1,20 @@
...
+            <f:entry title="${%Parsers}" description="${%title.description}">
+              <u:repeatable field="parsers"/>
+            </f:entry>
+          </f:advanced>
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/WarningsPublisher/global.jelly]: ...and added this one.

File: src/main/resources/hudson/plugins/warnings/WarningsPublisher/global.jelly [hunk 11]
```diff
@@ -1,18 +1,20 @@
...
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:u="/util"
+         xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+  <f:section title="${%Compiler Warnings}">
+      <j:choose>
+        <j:when test="${descriptor.canEditParsers()}">
+          <f:advanced title="${%Parsers}" align="left">
+            <f:entry title="${%Parsers}" description="${%title.description}">
+              <u:repeatable field="parsers"/>
+            </f:entry>
```
Comment (pjdarton) [inline src/main/resources/hudson/plugins/warnings/WarningsPublisher/global.jelly]: These 3 lines are unchanged aside from being indented by 2 spaces.

**Comment by uhafner** [comment]
Is there a way to get the advanced section automatically open if a new parser is added?

**Comment by uhafner** [comment]
I understand;-) Then we let your solution as it is for now. Extracting this piece of code is not complicated. I'll make this after the merge. (I need to adapt the merge for the new parser API as well, so I'll need to spend some time for this code anyway.)

**Pull request merged**